src/main/groovy/edu/oregonstate/mist/osuevents/resources/EventsResource.groovy

```diff
@@ -0,0 +1,239 @@
+            resultObject.data.attributes.each { key, value ->
+                updatedEvent."$key" = value
+            }
+        } catch (MissingPropertyException e) {
+            return badRequest(ErrorMessages.unknownFields).build()
+        } catch (Exception e) {
+            logger.error("Exception while calling updateEvent", e)
+            return internalServerError(ErrorMessages.unexpectedException).build()
+        }
+
+        //get custom fields and filters ready for DAO
+        String customFieldData = JsonOutput.toJson(updatedEvent.customFields)
+        String filterData = JsonOutput.toJson(updatedEvent.filters)
+
+        try {
+            updatedEvent.instances.each {
+                String start = it.start.toString()
+                String end = it.end.toString()
+
+                //if instance doesn't exist, create it
+                if (!eventsDAO.getInstance(id, it.id)) {
+                    eventsDAO.createInstance(
+                            it.id,
+                            id,
+                            InstanceMapper.formatForDB(start),
+                            InstanceMapper.formatForDB(end)
+                    )
+                //if start and end values are null, delete the instance
+                } else if ((start == "null") && (end == "null")) {
+                    eventsDAO.deleteInstance(id, it.id)
+                } else {
+                    eventsDAO.updateInstance(
+                            it.id,
+                            id,
+                            InstanceMapper.formatForDB(start),
+                            InstanceMapper.formatForDB(end)
+                    )
+                }
+            }
+        } catch (DateTimeParseException e) {
+            return badRequest(ErrorMessages.parseDate).build()
+        } catch (MissingMethodException e) {
+            return badRequest(ErrorMessages.processInstance).build()
+        } catch (Exception e) {
+            logger.error("Exception while calling updateEvent", e)
+            return internalServerError(ErrorMessages.unexpectedException).build()
+        }
+
+        eventsDAO.updateEvent(id, updatedEvent, filterData, customFieldData)
+
+        //get updated event and put it in response
+        ok(getResultObject(eventsDAO.getById(id))).build()
+    }
+/**
+ * DELETE an event
+ */
+    @DELETE
+    @Path('{id: [0-9a-zA-Z-]+}')
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response deleteEvent(@Auth AuthenticatedUser _,
+                                @PathParam('id') String id) {
+        if (!eventsDAO.getById(id)) {
+            return notFound().build()
+        }
+
+        eventsDAO.deleteEvent(id)
+        Response.noContent().build()
+    }
+
+/**
+ * Helper function for preparing a Result Object
+ */
+    private ResultObject getResultObject(def events) {
+        def resultObject = new ResultObject()
+        resultObject.data = []
+
+        events.each {
+            println(it.id)
```

Comment:
Extra println statement. You could change it to a log.debug statement so that it gets printed based on the debug level of the API.